Release/public v2 jet bugfix

natalie-perlin opened the pull request:
## DESCRIPTION OF CHANGES: 
1. Fixed the issue on running SRW on Jet: loading a specific version of miniconda3/4.5.12 from /contrib/., which was used to create a conda environment "regional_workflow". Prepending the path of /contrib/miniconda3/modulefiles to the $MODULEPATH and having it before the modulefiles hpc-stack directory did not solve the issue (see item 2. below, an earlier attempt)

2. Earlier attempt to solve:  load the miniconda3 from the /contrib/ modulefiles, not the module built with the HPC-stack. Use of "`module load -a /contrib/miniconda3/modulefiles/` " caused the module path to be appended to the search PATH, and the module path for the hpc-stack takes precedence when loading miniconda3. 

It did not help to solve the problem, because the miniconda3 in hpc-stack was labeled as a "default" module, regardless of the order in the MODULEPATH. Directory `/lfs4/HFIP/hfv3gfs/nwprod/hpc-stack/libs/modulefiles/core/miniconda3/` on Jet contains the following files:
```
4.6.14.lua
default -> 4.6.14.lua
```

## TESTS CONDUCTED: 

Making this PR to test if the issue with running SRW tests on Jet could be solved with fixing the correct miniconda3 module loaded, which needs to be the same module used to create the  environment 'regional_workflow'. 
Tests through Jenkins were successful:
[Jenkins-build-srw-jet-intel
](https://jenkins-epic.woc.noaa.gov/job/ufs-srweather-app/job/srw-build-intel/124/consoleFull)
[Jenkins-srw-e2e-tests
](https://jenkins-epic.woc.noaa.gov/job/ufs-srweather-app/job/srw-e2e-launch/135/consoleFull)

## ISSUE (optional): 
There were problems with running SRW tests on Jet after the recent HPC-stack update, with miniconda3 module becoming available in the hpc-stack build, as well as in /contrib/miniconda3/modulefiles.

## CONTRIBUTORS:
@mark-a-potts , @BruceKropp-Raytheon

Commit:
Updated versions of hpc-stack modules and compiler flags for building SRW release/public-v2

Commit:
Fix artifact creation for branches with / in name (#275)

The slash in branch names such as release/.* and feature/.* is
interpreted as a directory separator. This change replaces the "/"
character with a "_".

natalie-perlin committed:
Merge branch 'ufs-community:release/public-v2' into release/public-v2

natalie-perlin committed:
Corrections to the modulefiles for the MacOS

Modulefiles load software libraries and activate python virtual environment

natalie-perlin committed:
resolved conflict in modulefiles/build_macos_gnu

natalie-perlin committed:
correction in build_macos_gnu

natalie-perlin committed:
correction in build_macos_gnu

natalie-perlin committed:
Merge pull request #1 from ufs-community/release/public-v2

Release/public v2 UFS-COMMUNITY SRW repo - current

natalie-perlin committed:
Update wflow_jet

Module load miniconda3 from /contrib/...
Prepend module path /contrib/miniconda3/modulefiles/ to the PATH (not _append_)

JeffBeck-NOAA reviewed (approved):
@natalie-perlin, do you mind opening this same PR into the develop branch as well?  Thanks!

natalie-perlin committed:
Update wflow_jet

Adding a specific version of miniconda3/4.5.12 to be loaded, to avoid ambiguities with several available via /contrib/ and /hpc-stack/
